spec/factories/report.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :v2_report, class: 'V2::Report' do
+    to_create do |instance, context|
+      policy = FactoryBot.create(
+        :v2_policy,
+        :for_tailoring,
+        title: context.title,
+        system_count: context.assigned_system_count,
+        os_major_version: context.os_major_version,
+        compliance_threshold: context.compliance_threshold,
+        business_objective: context.business_objective,
+        account: context.account,
+        supports_minors: context.supports_minors
+      )
+
+      unsupported_systems = policy.systems.first(context.unsupported_system_count)
+      compliant_systems = (policy.systems - unsupported_systems).first(context.compliant_system_count)
+      other_systems = policy.systems - unsupported_systems - compliant_systems
+
+      unsupported_systems.each do |system|
+        FactoryBot.create(
+          :v2_test_result,
+          system: system,
+          account: context.account,
+          supported: 'f',
+          score: nil,
+          tailoring: V2::Tailoring.find_by!(policy: policy, os_minor_version: system.os_minor_version)
+        )
+      end
+
+      compliant_systems.each do |system|
+        FactoryBot.create(
+          :v2_test_result,
+          system: system,
+          account: context.account,
+          supported: 't',
+          score: 100,
+          tailoring: V2::Tailoring.find_by!(policy: policy, os_minor_version: system.os_minor_version)
+        )
+      end
+
+      other_systems.each do |system|
+        FactoryBot.create(
+          :v2_test_result,
+          system: system,
+          account: context.account,
+          supported: 't',
+          score: 10,
+          tailoring: V2::Tailoring.find_by!(policy: policy, os_minor_version: system.os_minor_version)
```

Comment:
I feel like the `tailoring` is redundant here if you know the `policy`. What about creating a transient attribute on the `TestResult` factory that would do the lookup for you? Then you can deduplicate...